### DISCUSSION TITLE
python-multipart package changed it's namespace, hence fastapi fails

### DISCUSSION THREAD

**Discussion by fingineering** [fastapi/fastapi#12532] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def ensure_multipart_is_installed() -> None:
    try:
        # __version__ is available in both multiparts, and can be mocked
        from multipart import __version__

        assert __version__
        try:
            # parse_options_header is only available in the right multipart
            from multipart.multipart import parse_options_header

            assert parse_options_header  # type: ignore[truthy-function]
        except ImportError:
            logger.error(multipart_incorrect_install_error)
            raise RuntimeError(multipart_incorrect_install_error) from None
    except ImportError:
        logger.error(multipart_not_installed_error)
        raise RuntimeError(multipart_not_installed_error) from None
```

### Description

* python-multipart changed its namespace from multipart to python_multipart in version 0.014, hence the package import check fails.
* installing python-mulitpart in version 0.0.12 helps
* It would probably help to either pin the version, or change the import of mulitpart

### Operating System

macOS

### Operating System Details

Should not matter which system

### FastAPI Version

0.115.3

### Pydantic Version

2.9.2

### Python Version

3.12.6

**Comment by saeedghadiri** [comment]
I tried using python-multipart==0.0.12 with FastAPI version 0.15.3, and while this prevented the initial error, it resulted in a 400 Bad Request error.

### Code to reproduce the issue:
```
from fastapi import APIRouter, File, UploadFile

router = APIRouter()

@router.post("/short")
async def perform_short(file: UploadFile = File(...)):
    audio_content = await file.read()
    return "ok"
```
### Expected behavior: 

The file should upload successfully without errors.

### Observed behavior:

- With FastAPI 0.15.3 and python-multipart==0.0.14: The application returns import error for python-multipart.
- With FastAPI 0.15.3 and python-multipart==0.0.12: It returns a 400 Bad Request error.
- With FastAPI 0.15.2: The issue is resolved.

**Reply by blueriver97** [reply]
I encountered a similar issue with OAuth, and in my case, it worked when I used python-multipart==0.0.13 with FastAPI version 0.115.3.

Although version 0.0.12 is the latest version available on PyPI, for some reason, version 0.0.13 was installed.

* MacOS 
* python 3.11.9 
* FastAPI 0.115.3

**Comment by BobbyTumur** [comment]
I just encountered with this issue on AsyncClient with pytest.
pytest kept failing, yelling me to install python-multipart when it's already installed as I checked **pip list**, the version was 0.0.15.
I tried uninstalling, installing it over and over again which didn't help me but found this link on stackoverflow:

[https://stackoverflow.com/questions/76659992/python-fastapi-package-python-multipart-installed-but-not-found](url)

and installed with "**pip install python-multipart==0.0.12**" which solved the problem.

I am not sure if it's related as I am complete noob, I don't even know what python-multipart does but got my problem fixed, so no complaint.

**Comment by odrling** [comment]
With the latest python-multipart `import multipart` should work [somehow](https://github.com/Kludex/python-multipart/pull/168) but `from multipart import __version__` doesn’t and causes the current check in FastAPI to fail.

There are two PRs trying to fix this issue currently:
https://github.com/fastapi/fastapi/pull/12607
https://github.com/fastapi/fastapi/pull/12618

**Reply by pawelad** [reply]
Related issue on the other repo: https://github.com/Kludex/python-multipart/pull/166

**Reply by Kludex** [reply, marked as answer]
It was an oversight. This was fixed on `python-multipart==0.0.16`.
